app/assets/javascripts/discourse/app/controllers/preferences/interface.js

```diff
@@ -2,19 +2,16 @@ import I18n from "I18n";
 import { inject } from "@ember/controller";
 import Controller from "@ember/controller";
 import { setDefaultHomepage } from "discourse/lib/utilities";
-import discourseComputed, { observes } from "discourse-common/utils/decorators";
-import {
-  listThemes,
-  previewTheme,
-  setLocalTheme
-} from "discourse/lib/theme-selector";
+import discourseComputed from "discourse-common/utils/decorators";
+import { listThemes, setLocalTheme } from "discourse/lib/theme-selector";
 import { popupAjaxError } from "discourse/lib/ajax-error";
 import {
   safariHacksDisabled,
   isiPad,
   iOSWithVisualViewport
 } from "discourse/lib/utilities";
 import { computed } from "@ember/object";
+import DiscourseEnv from "discourse-common/config/environment";
```

Comment:
@eviltrout I'm curious here, what should we be using?

```
Ember.testing
DiscourseEnv. environment
Discourse.testing
```

Is there anything which will play better with ember-cli ?

Reply:
Right now I am trying to figure out what test is hitting this, and see if it would be better to just write `location.reload` to do nothing :shrug:

Reply:
Yeah, we should avoid adding any test-specific code to the app. Instead, stub the `reload` function.

Reply:
@CvX I cant figure out how to stub `location.reload`. I tried with sandbox, and the property is unwriteable 
```
Promise rejected during "update some fields": Cannot redefine property: reload
```

Reply:
Right. We'll have to introduce a bit of indirection here: add a method (or a service) that does the `location.reload()` and stub that method/service.

Reply:
@CvX Alright a made a little helper called `page-reloader` which has a method we can stub :+1:

Reply:
I'm still curious about my initial question though :p because we have so many ways to check for testing

Reply:
I usually use `Ember.testing` but I am not sure yet if that breaks in Ember CLI. We should stick with that for now and if it doesn't work in the future we'll mass rename.